test/jasmine/tests/bar_test.js

```diff
@@ -397,6 +397,46 @@ describe('Bar.crossTraceCalc (formerly known as setPositions)', function() {
         assertArrayField(cd[2][0], 't.poffset', [-0.4]);
     });
 
+    it('should work with *width* typed arrays', function() {
+        var w = [0.1, 0.4, 0.7];
+
+        var gd = mockBarPlot([{
+            width: w,
+            y: [1, 2, 3]
+        }, {
+            width: new Float32Array(w),
+            y: [1, 2, 3]
+        }]);
+
+        var cd = gd.calcdata;
+        assertArrayField(cd[0][0], 't.barwidth', w);
+        assertArrayField(cd[1][0], 't.barwidth', w);
+        assertPointField(cd, 'x', [
+            [-0.2, 0.8, 1.8],
+            [0.2, 1.2, 2.2]
+        ]);
+    });
+
+    it('should work with *offset* typed arrays', function() {
+        var o = [0.1, 0.4, 0.7];
+
+        var gd = mockBarPlot([{
+            offset: o,
+            y: [1, 2, 3]
+        }, {
+            offset: new Float32Array(o),
+            y: [1, 2, 3]
+        }]);
+
+        var cd = gd.calcdata;
+        assertArrayField(cd[0][0], 't.poffset', o);
+        assertArrayField(cd[1][0], 't.poffset', o);
+        assertPointField(cd, 'x', [
+            [0.5, 1.8, 3.1],
+            [0.5, 1.8, 3.099]
```

Comment:
Huh, took me a while to figure out where these numbers came from... surprised I had never noticed this before but it seems weird that `offset` sets the left edge of each bar rather than its center. Oh well, a little late to change that. Why does the second one get `3.099` instead of `3.1`?

Reply:
I'm not sure. 

With a regular array, we get:

```
[0.5, 1.7999999999999998, 3.1]
```

with a typed array, we get:

```
[0.5000000014901161, 1.8000000059604644, 3.099999988079071]
```

Reply:
ah ok, that's just because it's a `Float32Array` ie single precision, not double. But it's still closer to 3.1 than to 3.099 😏